src/Arcus.Scripting.Sql/Scripts/Invoke-AzSqlDatabaseMigration.ps1

```diff
@@ -0,0 +1,154 @@
+
+# Get script to determine DB version
+$selectDbVersionScript = 'SELECT TOP 1 CurrentVersionNumber FROM DatabaseVersion ORDER BY CurrentVersionNumber DESC'
+
+#Create a DatabaseVersion table if it doesn't exist
+if(!$tables.ItemArray.Contains('DatabaseVersion'))
+{
+    Write-Host "The table 'DatabaseVersion' does not exist yet in this database."
+    Write-Host 'Creating DatabaseVersion Table...'
+    $createDbCommand = Get-SqlScriptFileText $ScriptsFolder 'CreateDatabaseVersionTable'
+    Execute-DbCommand $params $createDbCommand
+    Write-Host 'DatabaseVersion Table Created'
+
+    $databaseVersionNumber = 0
+}
+else
+{
+    # Get database versionnumber
+    $databaseVersionNumber = Get-CurrentDbVersionNumber
+}
+
+
+Write-Host "Current databaseVersionNumber : $databaseVersionNumber"
+
+#Run all necessary scripts
+$files = Get-ChildItem -Path $ScriptsFolder -Filter $ScriptsFileFilter | Sort-Object {$_.BaseName -replace "\D+" -as [Int]}
+
+for ($i=0; $i -lt $files.Count; $i++) 
+{
+    $fileName = $files[$i].BaseName
+    $fileNameSections = $fileName.Split('_');
+
+    if ($fileNameSections.Length -lt 2)
+    {
+        Write-Host "File $fileName skipped for not having all required name sections (version and description)."
+        continue;
+    }
+
+    $fileVersionStr = $fileNameSections[1];
+
+    [int] $scriptVersionNumber = -1;
+
+    if (-Not([int32]::TryParse($fileVersionStr, [ref]$scriptVersionNumber)))
+    {
+        Write-Host "File $fileName skipped because version is not valid."
+        continue;
+    }
+
+    [string] $scriptVersionDescription = [convert]::Tostring($fileNameSections[2])
+    if($scriptVersionDescription.Length -gt 256)
+    {
+        Write-Host "Need to truncate the migration description because its size is" $scriptVersionDescription.Length "while the maximum size is 256"
+        $scriptVersionDescription = $scriptVersionDescription.Substring(0,255)
+    }
+    
+    Write-Host "Found migration #$scriptVersionNumber with description '$scriptVersionDescription'"
+
+    if($scriptVersionNumber -eq ($databaseVersionNumber + 1))
+    {
+        Write-Host "Running script #$scriptVersionNumber"
+
+        # Perform migration
+        $text = Get-SqlScriptFileText $ScriptsFolder $fileName
+        Execute-DbCommand $params $text
+
+        # Append the new version and description in version table
+        $updateVersionQuery = "INSERT INTO [$DatabaseSchema].[DatabaseVersion] ([CurrentVersionNumber], [MigrationDescription]) VALUES ($scriptVersionNumber, '$scriptVersionDescription')"
+        Execute-DbCommand $params $updateVersionQuery
+
+        # Update DB version to new version
+        $databaseVersionNumber = $scriptVersionNumber
+
+        Write-Host "Migration to version $scriptVersionNumber complete."
+    }
+    else {
+        Write-Host "Migration #$scriptVersionNumber skipped"
+    }
+}
+
+#Get New Database Version Number
+$databaseVersionNumber = Get-CurrentDbVersionNumber
```

Comment:
@mbraekman , I think it's strange to call this again. The migrations get executed when there's script files with version numbers greater than the current DB version number; but the script ends with saying that it's still the 'old' version number.

Reply:
The purpose of this last call is to get the 'current' version after executing the migration.
Seeing as the "Get-CurrentDbVersionNumber"-function is executing the full SQL-script again to retrieve the latest version, it should actually retrieve the "new" version instead of the "old" one.
I'll give it a try to verify.

Reply:
Seems to be working correct:
![image](https://user-images.githubusercontent.com/16165565/109955958-5dea7900-7ce3-11eb-8f3e-aefc04825196.png)